File: csharp/Lift/LiftAndDoorPrinter.cs
using System;

namespace Lift
{
    public interface LiftPrinter
    {
        string printLiftForFloor(Lift lift, int floor);
    }

    public class LiftAndDoorPrinter : LiftPrinter
    {
        public string printLiftForFloor(Lift lift, int floor)
        {
            if (lift.Floor == floor)
                return printLift(lift, floor);
            else {
                String padding = LiftSystemPrinter.GetWhitespace(lift.Id.Length);
                if (lift.HasRequestForFloor(floor)) {
                    return "  *" + padding;
                } else {
                    return "   " + padding;
                }
            }
        }

        private string printLift(Lift lift, int floor)
        {
            if (lift.DoorsOpen) {
                if (lift.HasRequestForFloor(floor)) {
                    return "]*" + lift.Id + "[";
                } else {
                    return " ]" + lift.Id + "[";
                }
            }
            else {
                if (lift.HasRequestForFloor(floor)) {
                    return "[*" + lift.Id + "]";
                } else {
                    return " [" + lift.Id + "]";
                }
            }
        }
    }
}

File: java/src/test/java/codingdojo/LiftAndDoorPrinter.java
package codingdojo;

public class LiftAndDoorPrinter implements LiftPrinter {

    @Override
    public String printLiftForFloor(Lift lift, int floor) {
        if (lift.getFloor() == floor)
            return printLift(lift, floor);
        else {
            String padding = LiftSystemPrinter.getWhitespace(lift.getId().length());
            if (lift.hasRequestForFloor(floor)) {
                return "  *" + padding;
            } else {
                return "   " + padding;
            }
        }
    }

    private String printLift(Lift lift, int floor) {
        if (lift.areDoorsOpen()) {
            if (lift.hasRequestForFloor(floor)) {
                return "]*" + lift.getId() + "[";
            } else {
                return " ]" + lift.getId() + "[";
            }
        }
        else {
            if (lift.hasRequestForFloor(floor)) {
                return "[*" + lift.getId() + "]";
            } else {
                return " [" + lift.getId() + "]";
            }
        }
    }
}
